py/runtime.h

```diff
@@ -26,6 +26,7 @@
 #ifndef MICROPY_INCLUDED_PY_RUNTIME_H
 #define MICROPY_INCLUDED_PY_RUNTIME_H
 
+#include "py/mphal.h"
```

Comment:
This breaks the example/natmod builds... I think just remove it from here.  This header should only be included from .c files.

Reply:
This is tricky. Without a definition of `MICROPY_INTERNAL_WFE`, including `runtime.h` will fail (the difference between macros and inline functions is they force you to structure your headers with a more layered "include what you use"!)

Conceptually these "event" functions are a bit different, in as much as runtime.h is "pure Python runtime", mphal.h is "hardware abstractions", and then event functions are a bridge between the two.

I've updated this PR to put them into a new `py/event.h` header, which solves these kind of conflicts (and allows removing all the messing about with esp8266 headers, too.)

Let me know what you think.

Reply:
I'm pretty sure I tried just removing this include and the build worked.... but maybe I didn't test enough.

How about putting these inline functions in `py/mphal.h`?  As you say, they are half hardware abstraction functions.

Reply:
> How about putting these inline functions in `py/mphal.h`? As you say, they are half hardware abstraction functions.

Hmm, maybe that's a bad idea, because then `py/mphal.h` will need to include `py/runtime.h`, which is no good.

Reply:
Yeah, that's it. My very first version of this added the inlines to`mphal.h` and ran into that problem.

I think conceptually:

* `mphal.h` is "lower" and relates to the hardware.
* `runtime.h` is "higher" and relates to the pure Python layer.
* `event.h` is "in between" the two.